test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -327,6 +339,52 @@ top="test-compatibility-run"
 rm -rf "$top" && mkdir "$top"
 cd "$top"
 
+if [ "$upgrade_to_latest" = true ]; then
+    # Prepare test data.
+    echo "Preparing test data..."
+    git clone --quiet --depth 1 --filter=blob:none --no-checkout https://github.com/wiredtiger/mongo-tests.git
+    cd mongo-tests; git checkout --quiet master -- WT-8395; cd WT-8395; test_data=$(pwd)
+
+    for FILE in $test_data/*; do tar -zxf $FILE; done
+    rm *.tar.gz; cd ../..
+
+    test_root=$(pwd)
+    for b in ${upgrade_to_latest_upgrade_downgrade_release_branches[@]}; do
+        (build_branch $b)
+        cd $b/test/checkpoint
```

Comment:
Could we lift the contents of this for loop (L354-L383) into a function. More or less following the convention of the other tests in this file e.g. `run_format`/`verify_branches` (keeping this test modular).

Reply:
Great point! Moved the code into a new `test_upgrade_to_branch` function.